lib/ddtrace/tracer.rb

```diff
@@ -131,15 +131,15 @@ def configure(options = {})
       # Those are rare "power-user" options.
       statsd = options.fetch(:statsd, nil)
       sampler = options.fetch(:sampler, nil)
-      priority_sampling = options[:priority_sampling]
+      priority_sampling = options.fetch(:priority_sampling, true)
       max_spans_before_partial_flush = options.fetch(:max_spans_before_partial_flush, nil)
       min_spans_before_partial_flush = options.fetch(:min_spans_before_partial_flush, nil)
       partial_flush_timeout = options.fetch(:partial_flush_timeout, nil)
 
       @enabled = enabled unless enabled.nil?
       @sampler = sampler unless sampler.nil?
 
-      if priority_sampling
+      if priority_sampling && !@sampler.is_a?(PrioritySampler)
```

Comment:
Added this check because if you configured the tracer twice, it would wrap the previous `PrioritySampler` with another `PrioritySampler`, and that seems problematic...

Reply:
yeah, we have a weird thing in `dd-trace-py` where we have `priority_sampling` default to `nil` so that we only try to update the sampler if the option has been provided, rather than defaulting to `True`.

I think this basically does the same thing.

Reply:
it might be good to add a comment here about why we have this check here.

Reply:
Cool, I'll add that.

Reply:
I see we updated `options.fetch(:priority_sampling, true)`, does that mean we should have this be `if !priority_sampling.nil?` (sorry for my bad ruby)

Since `nil` is falsey, we probably only want to modify what sampler/writer we are using if we are setting priority sampling.


Also, do we need to handle `:priority_smapling => false`? if we reconfigure and set to `false` do we need to update the sampler/writer again to use a `RateSampler`?

Reply:
> Since `nil` is falsey, we probably only want to modify what sampler/writer we are using if we are setting priority sampling.

Consider the following: `{}.fetch(:a, true) #=> true`. Users would have to explicitly specify the key with a `nil` value in order to pass `nil`, in such a case I feel like we should treat it as `false` anyways.

Given the above, I think your original assertion is already the case here. The `if` block wouldn't run if given `nil`, but it should never receive `nil` unless given `nil` explicitly.

> Also, do we need to handle `:priority_smapling => false`? if we reconfigure and set to false do we need to update the sampler/writer again to use a `RateSampler`?

I think if we support this, then we need to add an `elseif`:

```ruby
elsif !priority_sampling
  @sampler = sampler || Datadog::AllSampler.new if @sampler.is_a?(PrioritySampler)
  @writer = Writer.new
end
```

Reply:
yeah, sorry I misread the code. In `dd-trace-py` we default to `None` (`nil`), which means the user did not supply the option so we then default to `True`.

How do we instruct people to use `#configure` ?

In `dd-trace-py` we allow people to supply only the arguments they want to modify.

An example:

```python
tracer.configure(priority_sampling=False)

tracer.configure(host='127.0.0.1')
```

Which will configure the tracer disabling priority sampling, and then configure the host to be localhost while keeping priority sampling disabled.


This might just be how we have `.configure()` setup for `dd-trace-py`, curious if this is (or should be) the same case here?

Reply:
Right, first call would deactivate, then second would reactivate. That's no good.

Changing the logic such that it defaults to `nil`, activates if `!= false`, and deactivates only if `== false`.